hive/glue-sync.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+source /usr/lib/hustler/bin/qubole-bash-lib.sh
+source /usr/lib/qubole/bootstrap-functions/hive/thrift-metastore.sh
+
+##
+# Installs Hive Glue Catalog Sync Agent
+# param1 - Region for AWS Athena. Defaults to us-east-1
+# Requires Hive 2.x
+#
+function install_glue_sync() {
+    aws_region=${1:-us-east-1}
+
+    is_master=$(nodeinfo is_master)
+    hive_version=$(nodeinfo hive_version)
+    if [[ ${is_master} == "1" && ${hive_version} == 2* ]]; then
+        glue_staging_dir=$(nodeinfo s3_default_location)
+        glue_staging_dir="${glue_staging_dir}/query_result"
+
+        hadoop dfs -get s3://paid-qubole/aws_glue_sync/HiveGlueCatalogSyncAgent-1.0-SNAPSHOT-jar-with-dependencies.jar /usr/lib/hive1.2/lib/
+
+        # Add glue sync configurations to hive-site.xml
+        # (Refer : https://github.com/awslabs/aws-glue-catalog-sync-agent-for-hive)
+        sed -i "s|</property></configuration>|</property><property><name>hive.metastore.event.listeners</name><value>com.amazonaws.services.glue.catalog.HiveGlueCatalogSyncAgent</value></property></configuration>|g" /usr/lib/hive1.2/conf/hive-site.xml
```

Comment:
Is it safe to assume this would always be in one line? Might be better to use an xml editor like xmlstarlet (though we would have to install that in the bootstrap as well for now - we can add it in the base ami later).

Reply:
Makes sense. Modified to use xmlstarlet instead of sed